code/modules/projectiles/ammunition/ballistic/harpoon.dm

```diff
@@ -0,0 +1,14 @@
+/obj/item/ammo_casing/harpoon
+	name = "harpoon"
+	caliber = CALIBER_HARPOON
+	icon_state = "magspear"
+	base_icon_state = "magspear"
+	projectile_type = /obj/projectile/bullet/harpoon
+
+/obj/item/ammo_casing/harpoon/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/caseless)
+
+/obj/item/ammo_casing/harpoon/update_icon_state()
+	. = ..()
+	icon_state = "[base_icon_state]"
```

Comment:
you definitely don't have to address this now but this pattern is repeated a lot for ammo casings, would be neat for another signal-listening element to be added that does this rather than have to do the proc override boilerplate. just mentioning now

Reply:
It's a problem with how icon states for casings are handled at the core. I was too thinking of adding a signal proc or element to stop icon states from being updated, but that may bite us back in a distant future anyway.